integrations/block-data-api.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Automattic\VIP\Integrations;
+
+/**
+ * Loads VIP Block Data REST API.
+ *
+ * @private
+ */
+class BlockDataApi extends Integration {
+	/**
+	 * Applies hooks to load Block Data API plugin.
+	 *
+	 * @private
+	 */
+	public function load( array $config ): void {
+		// Wait until plugins_loaded to give precedence to the plugin in the customer repo
+		add_action( 'plugins_loaded', function() {
+			// Do not load plugin if already loaded by customer code
+			if ( defined( 'VIP_BLOCK_DATA_API_LOADED' ) ) {
+				return;
+			}
+
+			$load_path = WPMU_PLUGIN_DIR . '/vip-integrations/vip-block-data-api-1.0.1/vip-block-data-api.php';
```

Comment:
Should this really be hardcoded? 

I.e. what happens when automation removes 1.0.1 (which currently doesn't exist because the sibling PR is not merged yet)?

Reply:
> Should this really be hardcoded?

@rinatkhaziev Not a big fan of this being hardcoded either. This implementation was somewhat inspired by the wp-parsely implementation that [loops through version paths](https://github.com/Automattic/vip-go-mu-plugins/blob/develop/wp-parsely.php#L271) based on the hardcoded [`SUPPORTED_VERSIONS`](https://github.com/Automattic/vip-go-mu-plugins/blob/develop/wp-parsely.php#L19-L27), which is a bit abstracted but still requires version string hardcoding.

Here are some ideas @ingeniumed and I have to solve this:

1. Move the `1.0.1` to a separate function or constant in `integrations/block-data-api.php`. This doesn't fix the hardcoding but makes the separation a bit cleaner.
2. Use [`glob()`](https://www.php.net/manual/en/function.glob.php) to match version paths and select the latest one. This may be a bit tricky as we'd need to parse out some sort of semantic versioning and comparison. I think there may be potential for performance penalties as well, but because this code is only run when the plugin is activated it's lower risk.

Any suggestions on how we can avoid hardcoding the path here? It would be great if we can remove that.

Reply:
What if we pass the version to activate as a part of `$config` on `activate`?

Reply:
Good question, we didn't go down that path because:

- We want to be the ones managing the version. We don't want to support version pinning via a constant, or via what someone else calling this method passes in. So calling this method would mean you get a version we define - which will currently be the latest one but in the future can be changed ([Alec posted this here, that touches on it](https://github.com/Automattic/vip-go-mu-plugins/pull/4406#issuecomment-1526996573)).
- We want this code to be generic, and not be directed at a specific plugin. In order to set the version [here in the config when `activate` is called](https://github.com/Automattic/vip-go-mu-plugins/blob/0efda5285de6b057d3f335468afabdfd0f4ae517/vip-integrations.php#L31), we would need to check what the plugin name is and set the version based on that. At the moment, there isn't really any specific code targeted at block data api within that linked code.

Given that, would option 1 be a good way to solve this? 

What that would involve:

- We define the version as a class level method [within the plugin integration similar to config](https://github.com/Automattic/vip-go-mu-plugins/blob/0efda5285de6b057d3f335468afabdfd0f4ae517/integrations/integration.php#L16) that a plugin can override. We would then call that in the load function. That would ensure that the version is exposed within the plugin itself, in a way that's more visible. We also add a note above the class level method to ensure that this version is carefully evaluated against the lowestVersion in mu-plugins-ext, and link each specific config.json within the implementation.
- Currently, if the file is not found the plugin is not loaded. We take that a step further and mark the active flag as false if the file is not found. We can add a test here to ensure that this is caught.

Would that be better? Thanks for the feedback!